src/nodes.coffee

```diff
@@ -1198,25 +1186,74 @@ exports.Class = class Class extends Base
     args  = []
     o.classScope = func.makeScope o.scope
 
+    superClass = null
+    if @parent
+      superClass = o.classScope.freeVariable 'superClass', reserve: no
+      func.params.push new Param new IdentifierLiteral superClass
+      args.push @parent
+
     @hoistDirectivePrologue()
     @setContext name
     @walkBody name, o
-    @ensureConstructor name
+    @ensureConstructor()
     @addBoundFunctions o
+    @ensureConstructorSuperCall() if @parent
+
+    jsClass = new JsClass name, superClass, @ctor
+    @hoistConstructorComment jsClass
+
     @body.spaced = yes
+    @body.expressions.unshift @directives..., jsClass.placeholder
     @body.expressions.push lname
+    @body.expressions.splice @body.expressions.indexOf(@ctor), 1, jsClass
 
-    if @parent
-      superClass = new IdentifierLiteral o.classScope.freeVariable 'superClass', reserve: no
-      @body.expressions.unshift new Extends lname, superClass
-      func.params.push new Param superClass
-      args.push @parent
+    klass  = new Parens new Call func, args
+    klass  = new Assign @variable, klass, null, { @moduleDeclaration } if @variable
+    result = klass.compileToFragments o
+    jsClass.placeholder.updateFragments result
+    result
+
+#### Javascript Classes
```

Comment:
I feel like we should call this “ECMAScript Classes” or `ESClass`. The phrase “JavaScript class” for me implies a `function`-based construct, like what’s generated by the CoffeeScript 1.x `class` keyword; as opposed to the ES2015+ output.